src/update_agent/actor.rs

```diff
@@ -16,6 +16,26 @@ lazy_static::lazy_static! {
         "zincati_update_agent_last_refresh_timestamp",
         "UTC timestamp of update-agent last refresh tick."
     )).unwrap();
+    static ref FINALIZATION_ATTEMPTS: IntCounter = register_int_counter!(opts!(
+        "zincati_update_agent_finalization_attempts",
+        "Total number of attempts to finalize a staged deployment by the update agent."
+    )).unwrap();
+    static ref FINALIZATION_ALLOWED_BY_STRATEGY: IntCounter = register_int_counter!(opts!(
+        "zincati_update_agent_finalization_allowed_by_strategy",
+        "Total number of finalization attempts by the update agent allowed by its update strategy."
+    )).unwrap();
+    static ref FINALIZATION_DENIED_BY_STRATEGY: IntCounter = register_int_counter!(opts!(
```

Comment:
These `denied` cases can also likely be folded into labels.

How do you plan to fit inhibitors into the flow (and thus metrics counting)?

Reply:
I was thinking of making a separate metric, similar to `zincati_update_agent_finalization_denied_by_usersessions` to count denials due to systemd inhibitors. Should we combine all non strategy-related gates into one?

Reply:
Yes, I'd combine them together via labels.